docs/next.config.js

```diff
@@ -111,22 +112,7 @@ module.exports = withDocsInfra({
                   [
                     'babel-plugin-module-resolver',
                     {
-                      alias: {
-                        // all packages in this monorepo
-                        '@mui/material': '../packages/mui-material/src',
-                        '@mui/docs': '../packages/mui-docs/src',
-                        '@mui/icons-material': '../packages/mui-icons-material/lib',
-                        '@mui/lab': '../packages/mui-lab/src',
-                        '@mui/styled-engine': '../packages/mui-styled-engine/src',
-                        '@mui/styles': '../packages/mui-styles/src',
-                        '@mui/system': '../packages/mui-system/src',
-                        '@mui/private-theming': '../packages/mui-private-theming/src',
-                        '@mui/utils': '../packages/mui-utils/src',
-                        '@mui/base': '../packages/mui-base/src',
-                        '@mui/material-next': '../packages/mui-material-next/src',
-                        '@mui/joy': '../packages/mui-joy/src',
-                      },
-                      // transformFunctions: ['require'],
+                      alias: getMuiAliases('src'),
```

Comment:
Now that aliases moved to webpack (through typescript paths), is this aliasing still necessary?

Reply:
Yes it seems. In Toolpad we don't use typescript paths, but do aliasing directly [in webpack](https://github.com/mui/mui-toolpad/blob/72c3c3cb7c30e6dfc96bb47f8f419d3d4ece7830/docs/next.config.js#L49-L53). I believe then it gets picked up under `node_modules` as well

Reply:
For the core these aliases are needed to make sure that when we use X components on the marketing werbsite, the `@mui/material` (& co) imports from `@mui/x-date-pickers` / `@mui/x-data-grid` are using the local one and not the one from their node modules (which causes issues).

But if we have a way to say to Next to also resolve some node modules with the TS aliases, that would be great

Reply:
> But if we have a way to say to Next to also resolve some node modules with the TS aliases, that would be great

Not with TS aliases I believe. But with webpack aliases it seems to work that way

Reply:
Not yet, indeed. But it could be an option to remove the aliases you've put in tsconfig, and define them in the webpack resolver under next.config.js instead. I believe that way even imports inside of node_modules will be resolved with these aliases. I've explored that option in this PR: https://github.com/mui/material-ui/pull/33199 and tested whether the datagrid still import the correct modules here https://github.com/mui/material-ui/pull/33199#discussion_r900518769

Reply:
> Not yet, indeed. But it could be an option to remove the aliases you've put in tsconfig, and define them in the webpack resolver under next.config.js instead

OK to go full webpack instead of full TypeScript for the aliases ?
I feel like the TypeScript aliases are required to run `tsc` correctly, but I would have to test.

Reply:
Interesting, that made me wonder why it currently is working then. It seems like we already define paths in the [top-level tsconfig](https://github.com/mui/material-ui/blob/c175f60bd3cf15f0fc55b754c990cffb47a30d47/tsconfig.json#L16-L43). Doesn't Next.js pick them already?

Reply:
> t seems like we already define paths in the [top-level tsconfig](https://github.com/mui/material-ui/blob/c175f60bd3cf15f0fc55b754c990cffb47a30d47/tsconfig.json#L16-L43).

Yes, and even with this PR we only define them their for TS
I just added some missing ones

I think that currently on master, babel resolves the aliases before NextJS has the opportunity to use the TS aliases
Or NextJS uses the TS aliases but some are missing and later resolved by babel (don't know what is the execution order).
But having both systems in parallel is weird.

Reply:
> having both systems in parallel

Could it be that the TS alias aren't meant for Next.js but the tests?

Reply:
> I think that currently on master, babel resolves the aliases before NextJS has the opportunity to use the TS aliases
Or NextJS uses the TS aliases but some are missing and later resolved by babel (don't know what is the execution order).

babel is the compiler which runs first, webpack is the linker, which links compiled sources together in a bundle. aliasing really only belongs in the linking phase. How it currently works is that next.js translates the tsconfig paths into webpack aliases (but ignores node_modules) and bundles the project using these aliases. Since babel runs first, it already does the aliasing and the ones that come from webpack (coming from tsconfig) do nothing since they're already replaced by babel.
Instead of letting next.js define the webpack aliases, we could define them ourselves, and include node_modules.


> Could it be that the TS aliases aren't meant for Next.js but for the tests?

I believe they are meant for typechecking.

Reply:
> How it currently works is that next.js translates the tsconfig paths into webpack aliases (but ignores node_modules) and bundles the project with webpack using these aliases. Since babel runs first, it already does the aliasing and the ones that come from webpack (coming from tsconfig) do nothing since they're already replaced by babel.

:+1: Make sense

Oops, I did not see #33199 

My main goal is to avoid duplication, so if we can get rid of the TypeScript aliases, that's even better for me
I'll try to play with it when I have some time.

Reply:
We probably won't be able to get rid of the typescript paths soon as it will be necessary for vscode. But I'd already consider it a win if we can remove them from `babel` everywhere.

Reply:
I struggle to apply the webpack aliases on the node_modules, if you have an idea on how to do it (to be able to drop babel which is necessary for notistack and the X deps)

Reply:
I struggle to apply the webpack aliases on the node_modules, if you have an idea on how to do it (to be able to drop babel which is necessary for notistack and the X deps)

Reply:
I updated https://github.com/mui/material-ui/pull/33199 and it still seems to work. Check out the changes to next.config.js without whitespace: https://github.com/mui/material-ui/pull/33199/files?diff=split&w=1#diff-97081d4c5e67d7fb92c87085affbac02322e2b4c53c01c5e65f500fc3b67a402

and babel.config.js: https://github.com/mui/material-ui/pull/33199/files?diff=split&w=1#diff-2ed4f5b03d34a87ef641e9e36af4a98a1c0ddaf74d07ce93665957be69b7b09a

Will give a quick try if I can replace `next-transpile-modules` with the upcoming built-in [`transpilePackages`](https://beta.nextjs.org/docs/api-reference/next.config.js#transpilepackages)

Reply:
OK but I guess it will only works with the doc, we will still have to keep babel for the testing.
That's still a win for sure.

Reply:
Yes, sorry for the confusion, I'm only talking about the docs here